addons/sale_crm/wizard/crm_opportunity_to_quotation_views.xml

```diff
@@ -17,7 +17,7 @@
                     </group>
                 </group>
                 <footer>
-                    <button name="action_apply" string="Confirm" type="object" class="btn-primary"/>
+                    <button name="action_apply" string="Confirm" type="object" class="btn-primary" invisible="1"/>
```

Comment:
@AntoineVDV 
we may remove it,  but from sale_renting_crm, we are also changing the method(name) of the button with view priority high as we have several cases as below which are not work in case of new structure proposal:

cases: 

- when only sale_management_crm installed>> open the quotation view

- when only sale_renting_crm installed  >> open the rental view(kanban)

- when sale_managment_crm and sale_renting_crm installed >>  open the renting view even though the module is installed in any sequence and if we just override the method then if sale_managment_crm is installed after the sale_renting_crm, it will always open the quotation view instead of a rental view.
can you please provide a suggestion on this?
Thanks

Reply:
@mba-odoo cc @ppr-odoo 
Good point! I didn't have the third case in mind when I wrote this review.

Ok then let's keep the new structure proposal and make some changes so that it works with all cases. I prefer a solution relying on inheritance rather than duplicating methods and renaming actions, although the later works.

So, we came up with the following design pattern that should do the job:
- We define a method returning a falsy (`{}` for instance) action in sale_crm.
- We inherit this method in sale_management_crm and make it return the result of `super` or, if the later is falsy, the action defined in sale_management_crm.
- We also inherit this method in sale_renting_crm and make it always return the action defined in sale_renting_crm.

With this pattern, we can handle the following cases:
- [x] If only sale_management_crm is installed: we return the action defined in sale_management_crm.
- [x] If only sale_management_crm is installed: we return the action defined in sale_renting_crm.
- [x] If both sale_management_crm and sale_renting_crm are installed and if sale_management_crm is loaded first: sale_renting_crm is at the end of the inheritance chain so we return the action defined in sale_renting_crm.
- [x] If both sale_management_crm and sale_renting_crm are installed and if sale_renting_crm is loaded first: sale_management_crm is at the end of the inheritance chain so we return the action defined in sale_renting_crm since the action returned by `super` is falsy.

In practice, the following changes should be made:
- Replace the current definition of `action_new_quotation` in sale_crm with this one:
  ```python
      def action_new_quotation(self):
      """
      Return a falsy action to be tested by inheriting modules and used as a beacon in the
      inheritance chain so that they don't need to rely on the (non-deterministic) loading order.        
      :rtype None
      """
      return None
  ```
- Add the inherited method `action_new_quotation` in sale_management_crm:
  ```python
     def action_new_quotation(self):
         action = super().action_new_quotation()
         if not action:
             action = self.env.ref("sale_management_crm.sale_action_quotations_new").read()[0]
             action['context'] = self._get_quotation_action_context()
         return action
  ```
- Move the definition of the action `sale_action_quotations_new` to sale_management_crm and take the opportunity to rename it to `new_quotation_action`.

I didn't test it myself so let me know if the solution has a design issue or if you have a better idea ;)

Reply:
Hello @AntoineVDV 

I've implemented the changes you suggested and it seems to be working as expected. However, in the latest implementation, there's one pitfall. If we only install sale_crm module, the button to create quotation will be visible on the crm lead view, but as we are returning None in the respective method instead of a real action dict, quotation won't be created. This might be confusing for end users. That is actually a rare case (usually sale_crm module will be installed as a dependency), but still, I want to make sure that you're aware of this case :)

So if you're okay with this, we can move forward with this implementation.

Reply:
Hello, you're absolutely right. This detail got lost in my message but you indeed need to keep the `invisible="1"` on the button definition in sale_crm and make it visible in sale_management_crm and sale_renting_crm.